includes/payment-methods/class-upe-payment-gateway.php

```diff
@@ -1133,12 +1150,8 @@ public function log_payment_error_ajax() {
 
 			$this->order_service->mark_payment_failed( $order, $intent_id, $intent_status, $charge_id, $error_message );
 
-			self::remove_upe_payment_intent_from_session();
-
 			wp_send_json_success();
 		} catch ( Exception $e ) {
-			self::remove_upe_payment_intent_from_session();
-
```

Comment:
I don't think we should delete `remove_upe_payment_intent_from_session` in these places, as `log_payment_error_ajax` handles errors happening during the confirmation stage of shoppers/buyers in the checkout page. 

IMO, as we will surely run this method in both `try` and `catch`, we should run it early, before try-catch.

Reply:
Inside try body: similar to https://github.com/Automattic/woocommerce-payments/pull/5653/files#r1122708333 - `$this->order_service->mark_payment_failed( $order, $intent_id, $intent_status, $charge_id, $error_message );` will trigger order status change hook and we'll perform intent cache reset.

The one in the catch statement needs to make sense - look carefully into exceptions potentially happening in try-statement.
Those are validations of posted data or errors getting intents/charges. The latest are read operations, so I don't see cases in there which do require the cache reset.

But I'm open to reviving this if there is a case in try we have a case requiring the cache cleaning.